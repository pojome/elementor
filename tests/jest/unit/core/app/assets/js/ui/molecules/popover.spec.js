import { render, fireEvent } from '@testing-library/react';
import Popover from 'elementor-app/ui/molecules/popover';

describe( '<Popover />', () => {
	it( 'should render the component', () => {
		// Arrange
		const text = 'Test children';

		// Act
		const component = render( <Popover> { text } </Popover> );

		// Assert
		expect( component.getByTestId( 'popover' ) ).toHaveTextContent( text );
	} );

	it( 'should call closeFunction prop when clicking on the popup', () => {
		// Arrange
		const mockFn = jest.fn();
		const component = render( <Popover closeFunction={mockFn} /> );

		// Act
		fireEvent.click( component.getByTestId( 'popover' ) );

		// Assert
		expect( mockFn ).toBeCalled();
	} );

	it( 'should call closeFunction prop when clicking on the background', () => {
		// Arrange
		const mockFn = jest.fn();
		const component = render( <Popover closeFunction={mockFn} /> );

		// Act
		fireEvent.click( component.getByTestId( 'background' ) );

		// Assert
		expect( mockFn ).toBeCalled();
	} );

	it( 'should match the snapshot', () => {
		// Arrange
		const { container } = render(
			<Popover className="test">
				<li> 1 </li>
				<li> 2 </li>
			</Popover>
		);

		// Act + Assert
		expect( container ).toMatchSnapshot();
	} );
} );
