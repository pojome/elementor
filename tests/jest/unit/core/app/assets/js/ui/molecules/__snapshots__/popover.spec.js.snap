// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popover /> should match the snapshot 1`] = `
<div>
  <div
    class="eps-popover__background"
    data-testid="background"
  />
  <ul
    class="eps-popover test"
    data-testid="popover"
  >
    <li>
       1 
    </li>
    <li>
       2 
    </li>
  </ul>
</div>
`;
